runtime/src/main/java/org/corfudb/protocols/wireprotocol/orchestrator/RemoveNodeWorkflow.java

```diff
@@ -0,0 +1,67 @@
+package org.corfudb.protocols.wireprotocol.orchestrator;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.Layout;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.corfudb.protocols.wireprotocol.orchestrator.OrchestratorRequestType.REMOVE_NODE;
+
+/**
+ * A workflow definition that removes a node from the cluster, if it exists. This workflow
+ * will fail if the remove results in a cluster that is not redundant.
+ * @author Maithem
+ */
+@Slf4j
+public class RemoveNodeWorkflow implements IWorkflow {
+
+    final RemoveNodeRequest request;
+
+    @Getter
+    final UUID id;
+
+    public RemoveNodeWorkflow(Request request) {
+        this.id = UUID.randomUUID();
+        this.request = (RemoveNodeRequest) request;
+    }
+
+    @Override
+    public String getName() {
+        return REMOVE_NODE.toString();
+    }
+
+    @Override
+    public List<Action> getActions() {
+        return Arrays.asList(new RemoveNode());
```

Comment:
Probably you want Collections.singletonList? Using Arrays here on a non-list is a bit odd

Reply:
Ok.